src/jsCodegen/runtimeLib.ts

```diff
@@ -23,22 +21,22 @@ export function determinePredicateTruthValue(iterator: { next: () => IterationRe
 // Make sure Closure Compiler does not change property names.
 declare interface IRuntimeLib {
 	adaptSingleJavaScriptValue: typeof adaptSingleJavaScriptValue;
+	atomize: typeof atomize;
 	determinePredicateTruthValue: typeof determinePredicateTruthValue;
 	DONE_TOKEN: typeof DONE_TOKEN;
 	getEffectiveBooleanValue: typeof getEffectiveBooleanValue;
 	isSubtypeOf: typeof isSubtypeOf;
 	ready: typeof ready;
-	XPDY0002: typeof XPDY0002;
 }
 
 const runtimeLib: IRuntimeLib = {
 	adaptSingleJavaScriptValue,
+	atomize,
```

Comment:
This atomize is a candidate to make a waaaaaaay simpler version of that just returns the atomized version of a node, in a simple JS object form. We can then use that in the normal `atomize` as well as in the runtime lib. That we we do not have to work with Sequences here: sequences are just a wrapper around iterators. There is a risk of the closure compiler renaming properties / methods of Sequence instances, causing errors that only occur in a built version of FontoXPath.

Reply:
Alternatively, for for instance attributes, you could consider checking what the type is of the code, if it is an attribute, compile `nodeValue ? nodeValue.data : ''`,  or something. And reject all other path expressions that you first need to atomize (those to elements, those to text nodes, etcetera)

Reply:
The atomize function has been replaced in the last two commits